RationaleMCP/0031/concept.md

```diff
@@ -0,0 +1,135 @@
+# Concept of Flat Modelica
+
+Flat Modelica is designed to be a standardized, intermediate file format which can be produced in the middle of the transformation process from a Modelica model to an executable model. 
+It might be used to support debugging but also by external tools to exploit the equations system further. 
+Therefore the core structure is a model containing all information to build up a mathematical equation system symbolically. All other information is moved to annotations. A tool can ignore them but might be still able to solve the equations. 
+
+## Design Rules 
+
+There are simple general rules for the design of Flat Modelica:
+
+* The syntax is a subset of Modelica syntax with some extensions.
+* Everything which has direct influence on the simulation results in an ideal mathematical sense has a special syntactical construct in the language.
+* Information which has no direct influence to the result are handeled uniformly by one syntactical construct called annotation. For example, this applies to
+   * prefixes
+   * attributes of variables 
+   * information lost in preceding transformation steps,
+   * information to control further transformation steps,
+   * information to support the numerical solution of the equation system
+   * source code location information, 
+   * other data for special use cases, e.g. packaging into a FMU, inputs and outputs of the whole model.
+* Description strings (also called syntactic comments) are an exception. These are optionally copied as such.
+* An external tool does not need any knowledge of special Modelica semantics, e.g. handling start attributes, behavior of relations and built-in operators for events,etc..
+
+## Details
+
+* All higher level syntactical and structural contructs of Modelica are removed, such as:
+   * imports
+   * inheritance
+   * redeclarations
+   * nested class definitions (There is no class definition inside another class definition)
+   * packages, [expandable] connectors, models (only one model remains), blocks, operator definitions
+   * functional inputs of functions
+   * conditional declarations
+   * modifications (therefore no merging of them)
+   * connect statements
+   * loops inside equation sections
+   * `if`/`then`/`else` in equation sections (atleast with different sized branches)
+   * `when` clauses
+   * stream operators: `actualStream(...)`, `inStream(...)`
+   * all prefixes, except `input` and `output`
+* __TODO__ synchronous Modelica extensions, including state machines, possibly introduction of partitions
+
+### Names
+
```

Comment:
I think we should avoid duplicating https://github.com/modelica/ModelicaSpecification/blob/MCP/0031/RationaleMCP/0031/name-mapping.md here.  If there is more to say about naming, I think it should go into that file.

Reply:
Agree.